File: app/src/main/java/com/botijasoftware/utils/Atlas.java
package com.botijasoftware.utils;

public class Atlas {

	public Atlas(Texture texture) {
		mTexture = texture;
	}
	
	public void LoadContent(ResourceManager resources) {
	}
	
	public int getTextureID() {
		return mTexture.getID();
	}
	
	public void flipCoords() {
	
		for (int i = 0; i < mNumTextures; i++) {
			mCoords[i].flipVertical(); 
		}
	}
	
	public int getTextureCount() {
		return mNumTextures;
	}
	
	
	public TextureCoords getTextureCoords(int textLinear) {
		return mCoords[textLinear];
	}

	public TextureRegion getAsTexture(int textLinear) {
		return new TextureRegion(mTexture, getTextureCoords(textLinear));
	}
	
	
	protected Texture mTexture;
	protected TextureCoords[] mCoords;
	protected int mNumTextures;
}

File: app/src/main/java/com/botijasoftware/utils/Atlas.kt
package com.botijasoftware.utils

open class Atlas(var mTexture: Texture) {

    protected lateinit var mCoords: Array<TextureCoords>
    var textureCount: Int = 0

    fun LoadContent(resources: ResourceManager) {}

    val textureID: Int
        get() = mTexture.id

    fun flipCoords() {

        for (i in 0..textureCount - 1) {
            mCoords[i].flipVertical()
        }
    }


    fun getTextureCoords(textLinear: Int): TextureCoords {
        return mCoords[textLinear]
    }

    fun getAsTexture(textLinear: Int): TextureRegion {
        return TextureRegion(mTexture, getTextureCoords(textLinear))
    }

}